src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookExplorerViewlet.ts

```diff
@@ -314,6 +320,10 @@ export class NotebookExplorerViewPaneContainer extends ViewPaneContainer {
 				this.searchWidget.focus(false, true); // focus back to input field
 			}
 		}, onQueryValidationError);
+		let end = new Date().getTime();
+		this._telemetryService?.createActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.NbTelemetryAction.SearchComplete)
```

Comment:
1. This isn't fired when the search is complete - this function just kicks off the search and returns immediately. The actual search will complete at some point later
2. Typically we use camelCase for property names (`timeTaken`)
3. Time taken should be a measurement - not a property
4. All measurements should have their units in the name (such as `timeTakenMs`)